### PR TITLE
Rooms API

### PR DESCRIPTION
**Pull request by ivanovaleksey**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -25,13 +37,11 @@ fn main() {
         .set_client_id("signals-rs");
```
Comment (manifest) [inline src/main.rs]: invalid client_id

File: src/messages.rs [hunk 2]
```diff
@@ -1,14 +1,95 @@
...
+    RoomsCreateRequest(RoomsCreateRequest),
+    RoomsCreateResponse(RoomsCreateResponse),
+    RoomsListRequest(RoomsListRequest),
+    RoomsListResponse(RoomsListResponse),
 }
 
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct RoomsCreateRequest {
+    pub payload: RoomsCreateRequestPayload,
+    cid: String,
```
Comment (manifest) [inline src/messages.rs]: Wouldn't it better to use string slices `cid: &'a str`?

File: src/messages.rs [hunk 3]
```diff
@@ -1,14 +1,95 @@
...
+
+type RoomsCreateResponseData = RoomsCreateRequestPayload;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct RoomsListRequest {
+    cid: String,
+}
+
+impl RoomsListRequest {
+    pub fn build_response(self, rooms: &Vec<models::Room>) -> RoomsListResponse {
```
Comment (manifest) [inline src/messages.rs]: Isn't it a RoomsListResponse constructor?

File: src/main.rs [hunk 4]
```diff
@@ -57,29 +67,72 @@ fn main() {
...
+            Message::Ping => {
+                let payload = serde_json::to_string(&Message::Pong).unwrap();
+                Ok(client.publish("pong", QoS::Level0, payload.into_bytes())?)
             }
-        }
-        _ => unimplemented!(),
+            _ => unimplemented!(),
+        },
+        Topic::Pong => unreachable!(),
+        Topic::Agent(t) => match t.room {
```
Comment (manifest) [inline src/main.rs]: Maybe we sould have some kind of controllers for nested resources. Fist, match entire topic against Rooms pattern -> CRUDLController::Rooms (implements CRUDL operations on rooms). Then, match room topic against Tracks pattern -> CRUDLController::Rooms (implements CRUDL operations on tracks).

File: src/messages.rs (hunk 2 above)
Comment (ivanovaleksey) [reply src/messages.rs]: Not sure this can be done, see https://github.com/serde-rs/serde/issues/359

File: src/main.rs (hunk 1 above)
Comment (ivanovaleksey) [reply src/main.rs]: We are going to fetch from `ENV` in the future

**Pull request merged**